pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -787,3 +789,147 @@ def longest_common_subsequence(seq1: OneDimensionalArray, seq2: OneDimensionalAr
+    : This modified implementation adds a couple of parameters, left and 
+    right, that indicate which portion of the array should be sorted. This
+    allows the Timsort algorithm to sort a portion of the array in place. 
+    Modifying the function instead of creating a new one means that it can 
+    be reused for both insertion sort and Timsort.
+    
+  
+    '''
+    
+    if right is None:
+        right = len(array) - 1
+
+    # Loop from the element indicated by
+    # `left` until the element indicated by `right`
+    for i in range(left + 1, right + 1):
+        # This is the element we want to position in its
+        # correct place
+        key_item = array[i]
+
+        # Initialize the variable that will be used to
+        # find the correct position of the element referenced
+        # by `key_item`
+        j = i - 1
+
+        # Run through the list of items (the left
+        # portion of the array) and find the correct position
+        # of the element referenced by `key_item`. Do this only
+        # if the `key_item` is smaller than its adjacent values.
+        while j >= left and array[j] > key_item:
+            # Shift the value one position to the left
+            # and reposition `j` to point to the next element
+            # (from right to left)
+            array[j + 1] = array[j]
+            j -= 1
+
+        # When you finish shifting the elements, position
+        # the `key_item` in its correct location
+        array[j + 1] = key_item
+
+    return array
+
+def timsort(array):
+    '''
+   
+    The Timsort algorithm is considered a hybrid sorting algorithm because
+    it employs a best-of-both-worlds combination of insertion sort and 
+    merge sort. The main characteristic of Timsort is that it takes 
+    advantage of already-sorted elements that exist in most real-world 
+    datasets.These are called natural runs. The algorithm then iterates
+    over the list, collecting the elements into runs and merging them into 
+    a single sorted list.
+    
+    
+    Parameters
+    ========
+
+    array: Array
+        The required array to be sorted
+    left: int
+        The starting index of the portion
+        which is to be sorted.
+        Optional, by default 0
+    right: int
+        The ending index of the portion which
+        is to be sorted.
+        Optional, by default the index
+        of the last position filled.
+
+    Returns
+    =======
+
+    output: Array
+        The sorted list or array
+
+    Examples
+    ========
+    
+    >>> from pydatastructs import OneDimensionalArray, timsort
+    >>> arr = OneDimensionalArray(int,[-2, 7, 15, -14, 0, 15, 0,  
+           7, -7, -4, -13, 5, 8, -14, 12] )
+    >>> timsort(arr)
+    >>> [arr[0], arr[1], arr[2], arr[3], arr[4], arr[5], arr[6], 
+        arr[7], arr[8], arr[9], arr[10], arr[11], arr[12], arr[13], 
+        arr[14]]
+    [-14, -14, -13, -7, -4, -2, 0, 0, 5, 7, 7, 8, 12, 15, 15] 
```

Comment:
This is not the correct way of adding docs. See examples of the above functions for reference